linera-core/src/worker.rs

```diff
@@ -611,7 +613,7 @@ where
             CertificateValue::ValidatedBlock {
                 executed_block: ExecutedBlock { block, .. },
             } => block,
-            CertificateValue::ConfirmedBlock { .. } => panic!("Expecting a validation certificate"),
+            _ => panic!("Expecting a validation certificate"),
```

Comment:
Using a catch all here means that if a variant is added to `CertificateValue` (maybe `Timeout`?) the compiler won't force you to explicitly handle this case.

Reply:
Yes, but I'd say that's conceptually correct here, since `process_validated_block` is only called with validated blocks.

Actually, I'd like to make this type safe in the future, and introduce separate types for confirmed and validated blocks: There are a lot of places where we _know_ we have a particular kind of certificate value, but we currently have no type for it.

Reply:
`_` is ok  if we know that the branch will likely apply to future variants.
In doubt, don't use `_` so that the compiler walks you through the necessary changes in the future.

Reply:
Yes, since this is `process_validated_block` I think it's safe to assume we'll always expect the `ValidatedBlock` variant here.